compiler/src/dmd/func.d

```diff
@@ -4558,7 +4558,7 @@ bool setUnsafe(Scope* sc,
                 .error(loc, fmt, arg0 ? arg0.toChars() : "", arg1 ? arg1.toChars() : "", arg2 ? arg2.toChars() : "");
                 return true;
             }
-            else if (!(sc.varDecl.storage_class & STC.system))
+            else if (!(sc.varDecl.storage_class & STC.trusted))
```

Comment:
The safety has 4 states:
1. none
2. system
3. trusted
4. safe

The trouble here is what if the safe bit is set? This will set both the safe and the system bit, which is an invalid state. What is necessary is not if it is not in the trusted state, but if it is in the default or safe state, the bits should be set to system.
Ideally, these bits for the 4 states should be replaced with an enum, but that's refactoring and should be a separate PR.

Reply:
With an enum, an invalid state will no longer be possible.

Reply:
> The trouble here is what if the safe bit is set?

There's a branch directly above for when the safe bit is set.

Reply:
Thank you, I didn't see that.